ably/rest_client_test.go

```diff
@@ -187,22 +188,53 @@ func TestRestClient(t *testing.T) {
 		}
 		res.Body.Close()
 
-		longAgo := lastInterval.Add(-120 * time.Minute)
-		page, err := client.Stats(&ably.PaginateParams{
-			Limit: 1,
-			ScopeParams: ably.ScopeParams{
-				Start: ably.Time(longAgo),
-				Unit:  proto.StatGranularityMinute,
-			},
-		})
-		if err != nil {
+		statsCh := make(chan []*proto.Stats, 1)
+		errCh := make(chan error, 1)
+
+		go func() {
```

Comment:
used a goroutine here so that I can return out of the for-select loop, rather than use `break label`

Reply:
I actually think a `break` would make this a little more concise:

```go
// poll the stats endpoint for 10 seconds until we get some stats
var stats []*proto.Stats
timeout := 10 * time.Second
for start := time.Now(); time.Since(start) < timeout; time.Sleep(time.Second) {
	longAgo := lastInterval.Add(-120 * time.Minute)

	page, err := client.Stats(&ably.PaginateParams{
		Limit: 1,
		ScopeParams: ably.ScopeParams{
			Start: ably.Time(longAgo),
			Unit:  proto.StatGranularityMinute,
		},
	})
	if err != nil {
		ts.Fatal(err)
	}
	if s := page.Stats(); len(s) > 0 {
		stats = s
		break
	}
}
if len(stats) == 0 {
	ts.Fatal("failed to get stats after %s", timeout)
}
```